subprojects/ide/src/integTest/groovy/org/gradle/plugins/ide/eclipse/EclipseIntegrationTest.groovy

```diff
@@ -43,7 +43,7 @@ class EclipseIntegrationTest extends AbstractEclipseIntegrationTest {
     void canCreateAndDeleteMetaData() {
         when:
         File buildFile = testFile("master/build.gradle")
-        usingBuildFile(buildFile).withTasks("eclipse").run()
+        executer.inDirectory(file("master")).withTasks("eclipse").run()
```

Comment:
Can not do this - `master` subdirectory is where the settings file resides for this project under test: https://github.com/gradle/gradle/tree/9c89f98a4e1ed5266c71b9f3349f9b2f404567f9/subprojects/ide/src/integTest/resources/org/gradle/plugins/ide/eclipse/EclipseIntegrationTest/canCreateAndDeleteMetaData
Perhaps the test project can be restructured to a standard structure with settings file at the root, but that I'd rather handle outside the scope of deprecations.

Reply:
If I understand correctly, with the special handling of `master` directory that we deprecated and removed in 7.0 https://github.com/gradle/gradle/issues/14398 your suggestion would have worked as Gradle would then look for a settings file in `master` subdirectory.

Reply:
So we still use the master directory feature in our integration tests. I think we should not do that unless it is to test the feature itself. I think opening an issue (and assigning it to 8.0-rc-1) is enough for now.

Reply:
Created a follow-up issue: https://github.com/gradle/gradle/issues/16735
I think it makes sense to handle this nowadays because those tests use `includeFlat` which we intend to deprecate now anyway https://github.com/gradle/gradle/issues/13891